Project/Models/bi_lstm_model.py

```diff
@@ -0,0 +1,115 @@
+        x = layers.Bidirectional(layers.LSTM(64))(x)
+        # Add a classifier
+        outputs = layers.Dense(1, activation="sigmoid")(x)
+        model = keras.Model(inputs, outputs)
+        print("Model created...!!!")
+        return model
+
+    # Method for training the model
+    def train_model(self, model):
+        print("Training Model... Please wait!")
+        model.compile("adam", "binary_crossentropy", metrics=["accuracy"])
+        model.fit(self.__x_train, self.__y_train, batch_size=32, epochs=2, validation_data=(self.__x_val, self.__y_val))
+        print("Model trained...!!!")
+        return model
+
+    # Method for saving trained model
+    def save_model(self, model):
+        path = os.path.dirname(__file__)
+
+        save_nn_model(model, path, "bi_lstm_model")
+        print("Model saved...")
+
+    # Method for loading saved model
+    def load_model(self):
+        filename = path = os.path.dirname(__file__) + "/bi_lstm_model"
+
+        loaded_model = load_nn_model(filename)
+        print("Model loaded successfully...")
+
+        return loaded_model
+        
+
+    def execute(self):
+        try:
+            # Load the saved model
+            model = self.load_model()
+
+            # Try with new review
+            new_review = ["""I absolutely adored this movie. For me, the best reason to see it is how stark a contrast it is from legal dramas like "Boston Legal" or "Ally McBeal" or even "LA Law." This is REALITY. The law is not BS, won in some closing argument or through some ridiculous defense you pull out of your butt, like the "Chewbacca defense" on South Park.) This is a real travesty of justice, the legal system gone horribly wrong, and the work by GOOD lawyers - not the shyster stereotype, who use all of their skills to right it. It will do more for restoring your faith in humanity than any Frank Capra movie or TO KILL A MOCKINGBIRD. And most importantly, I wept. During the film, during the featurette included at the end of the DVD - it's amazing. Wonderful film; wonderfully made. Thank God the filmmakers made it."""]
+            word_indices = imdb.get_word_index()
+            reviews = []
+            for doc in new_review:
+              review = []
+              for word in doc:
+                if word not in word_indices:
+                  review.append(2)
+                else:
+                  review.append(word_indices[word] + 3)
+              review.sort(reverse=True)
+              reviews.append(review)
+            x_test = keras.preprocessing.sequence.pad_sequences(reviews, truncating = 'post', padding = 'post', maxlen = self.__maxlen)
+            #print(model.predict(x_test))
+
+            self.predict(model, new_review)
+
+        except:
+            print("Unexpected error:", sys.exc_info()[0:2])
+            
+    def predict_reviews(self, raw_data):
+        model = self.load_model()
+        return self.predict(model, data)
+

```

Comment:
Are you sure, you tested this part because, getting raw_data and sending data further..